src/extendedstorage.cpp

```diff
@@ -671,10 +672,18 @@ void ExtendedStorage::setAlarms( const Incidence::Ptr &incidence )
       }
       e.setAttribute( "type", "todo" );
     } else if ( incidence->dtStart().isValid() ) {
-      e.setAttribute( "time", incidence->dtStart().toString() );
-      e.setAttribute( "startDate", incidence->dtStart().toString() );
-      if (incidence->endDateForStart( incidence->dtStart()).isValid() ) {
-        e.setAttribute( "endDate", incidence->endDateForStart( incidence->dtStart()).toString() );
+      KDateTime eventStart;
+
+      if (incidence->recurs()) {
+        // assuming alarms not later than event start
+        eventStart = incidence->recurrence()->getNextDateTime( alarmTime.addSecs(-60) );
+      } else {
+        eventStart = incidence->dtStart();
+      }
+      e.setAttribute( "time", eventStart.toString() );
+      e.setAttribute( "startDate", eventStart.toString() );
+      if (incidence->endDateForStart( eventStart).isValid() ) {
+        e.setAttribute( "endDate", incidence->endDateForStart( eventStart ).toString() );
```

Comment:
Inconsistent formatting
* line 685: if (xx ), most format either if (xx) or if ( xx ) to be in line with the rest of the file, the latter seems to be used more.
* line 685 uses ( xx).xx() and 686 ( xx ).xx(). The latter seems to be used more in this file.

Both bullet points applies to the duplicated code starting from line 941 or so.